plugin/httptrace/clienttrace.go

```diff
@@ -41,116 +40,147 @@ var (
 
 type clientTracer struct {
 	context.Context
-	httptrace.ClientTrace
 
 	tr trace.Tracer
 
-	levels map[string]trace.Span
-	root   trace.Span
-	mtx    sync.Mutex
+	activeHooks map[string]trace.Span
+	root        trace.Span
+	mtx         sync.Mutex
 }
 
-func newClientTracer(ctx context.Context) *clientTracer {
+func NewClientTrace(ctx context.Context) *httptrace.ClientTrace {
 	ct := &clientTracer{
-		Context: ctx,
-		levels:  make(map[string]trace.Span),
+		Context:     ctx,
+		activeHooks: make(map[string]trace.Span),
 	}
+
 	ct.tr = global.TraceProvider().GetTracer("go.opentelemetry.io/otel/plugin/httptrace")
-	ct.open("http.request")
-	return ct
+	ct.start("http.request", "http.request")
+
+	return &httptrace.ClientTrace{
+		GetConn:              ct.getConn,
+		GotConn:              ct.gotConn,
+		PutIdleConn:          ct.putIdleConn,
+		GotFirstResponseByte: ct.gotFirstResponseByte,
+		Got100Continue:       ct.got100Continue,
+		Got1xxResponse:       ct.got1xxResponse,
+		DNSStart:             ct.dnsStart,
+		DNSDone:              ct.dnsDone,
+		ConnectStart:         ct.connectStart,
+		ConnectDone:          ct.connectDone,
+		TLSHandshakeStart:    ct.tlsHandshakeStart,
+		TLSHandshakeDone:     ct.tlsHandshakeDone,
+		WroteHeaderField:     ct.wroteHeaderField,
+		WroteHeaders:         ct.wroteHeaders,
+		Wait100Continue:      ct.wait100Continue,
+		WroteRequest:         ct.wroteRequest,
+	}
 }
 
-func (ct *clientTracer) open(name string, attrs ...core.KeyValue) {
-	_, sp := ct.tr.Start(ct.Context, name, trace.WithAttributes(attrs...), trace.WithSpanKind(trace.SpanKindClient))
+func (ct *clientTracer) start(hook, spanName string, attrs ...core.KeyValue) {
+	_, sp := ct.tr.Start(ct.Context, spanName, trace.WithAttributes(attrs...), trace.WithSpanKind(trace.SpanKindClient))
+	// TODO(paivagustavo): remove this for loop when `trace.WithAttributes(attrs...)` works.
```

Comment:
Doesn't it already work in master? Also, if it doesn't, you could use `sp.SetAttributes()`.

Reply:
Yup I removed the TODO and fixed the bug :)